api/environments/models.py

```diff
@@ -174,8 +175,36 @@ def clone(self, name: str, api_key: str = None) -> "Environment":
         # Since identities are closely tied to the environment
         # it does not make much sense to clone them, hence
         # only clone feature states without identities
-        for feature_state in self.feature_states.filter(identity=None):
-            feature_state.clone(clone, live_from=feature_state.live_from)
+        queryset = self.feature_states.filter(identity=None)
+
+        if self.use_v2_feature_versioning:
+            # Grab the latest feature versions from the source environment. Note that,
+            # to clone these later on in the logic, we need the ORM objects, not the
+            # RawSQL queryset, so we have to nest the queries here.
+            latest_environment_feature_versions = EnvironmentFeatureVersion.objects.filter(
+                uuid__in=[
+                    efv.uuid
+                    for efv in EnvironmentFeatureVersion.objects.get_latest_versions(
+                        self
+                    )
+                ]
+            )
```

Comment:
I'm not particularly happy with this, but the alternative was to update the sql to return all the fields and then manually create the EnvironmentFeatureVersion, rather than using a `clone` method on the model class. To keep things consistent with other models, this seemed like the lesser of 2 evils. 

One option might be to conceal this in another method on the EnvironmentFeatureVersion model manager.